### DISCUSSION TITLE
1.5.1.4

### DISCUSSION THREAD

**Discussion by ilyhalight** [ilyhalight/voice-over-translation#575] [Обновления]
Все изменения были внесены @SashaXser в #560 и #573

## Список изменений
- Добавлено автоматическое определение языка для перевода прямых ссылок
- Добавлено обновление субтитров после перевода видео
- Добавлено сохранение перевода при изменение качества видео
- Возвращен addExtraEventListener emptied взамен на lipsync emptied
- Убрана активация скрипта с локальных файлов .mp4 (они не переводятся)

<hr /><em>This discussion was created from the release <a href='https://github.com/ilyhalight/voice-over-translation/releases/tag/1.5.1.4'>1.5.1.4</a>.</em>

**Comment by NullVerdict** [comment]
Userscripts не запускает скрипт из-за [GM_addStyle](https://github.com/ilyhalight/voice-over-translation/issues/571)

**Reply by ilyhalight** [reply]
Добавил уменьшенный [gm4-polyfill](https://github.com/greasemonkey/gm4-polyfill/blob/master/gm4-polyfill.js), где остался только GM_addStyle. Проблема должна уйти

**Reply by NullVerdict** [reply]
GM_addStyle не использовался ранее, он появился как модуль webpack-monkey. Вроде как его можно спойкойно удалить.

**Reply by ilyhalight** [reply]
> GM_addStyle не использовался ранее, он появился как модуль webpack-monkey. Вроде как его можно спойкойно удалить.

webpack-monkey [меняет](https://github.com/ilyhalight/voice-over-translation/blob/1dd21639493cf30ed1d12572cad7a0e10c1e3eb7/dist/vot.user.js#L538) логику загрузчика стилей на свою. Если просто удалить строчку с `// @grant GM_addStyle`, то будет выдавать ошибку как у человека в https://github.com/ilyhalight/voice-over-translation/issues/571#issuecomment-2028080035. Возможно, как-то можно отключить переопределение логики загрузчика стилей не меняя код webpack-monkey, но я не нашел как.
![изображение](https://github.com/ilyhalight/voice-over-translation/assets/62353659/7bbe2c27-1f30-44f1-9de5-7144a32dc290)

**Comment by ilya-corneli** [comment]
rumble перестал показывать панель перевода.

**Reply by NullVerdict** [reply]
Попробуй [фикс](https://github.com/SashaXser/voice-over-translation/tree/test/dist), должно исправить проблему

**Comment by Lexx0109** [comment]
Как можно увеличить текст субтитров?

**Comment by FiLinX** [comment]
вчера случайно удалил темпермонку, но затем тут же поставил её вновь и подтянул в ней последний мой бекап, - но оказалось что настройка обновления на VOT скрипте стала включенной, и скрипт со стоявшего у меня 1.5.1.3 обновился до 1.5.1.4  -  и обнаружилась вернувшаяся проблема с отсутствием панельки на ютубных эмбедах ((
пока откатился на 1.5.1.3

**Reply by NullVerdict** [reply]
Попробуй [фикс](https://github.com/SashaXser/voice-over-translation/tree/test/dist), может быть проблема пропадет

**Reply by FiLinX** [reply]
попробовал  -  есть панелька на эмбедах!
Спасибо!
правда странно скрипт обновило  -  в редакторе темпермонки он весь в одну строку стал)  -  https://i.imgur.com/g5LyIoy.png
хотя, может быть это из-за необычной установки так..
ну, я вот выше в посте кликнул на **фикс**, открылась страница с версиями скрипта, я кликнул на привычную мне vot.user.js, открылась страница просмотра его кода, ну и вот там уже я кликнул в его панели на таб Raw - появилось окно темпермонки с предложением обновить, я и обновил)

**Reply by FiLinX** [reply]
я всё же решил переустановить по нормалу..
сначала вновь откатил на 1.5.1.3, затем скачал выше с фикса vot.user.js и накатил уже его перетянув файл на окно темпермонки  -  теперь код скрипта там удобоваримо-многострочный)
эмбеды работают

**Comment by FiLinX** [comment]
во млин, впервые увидел в увидомлении на панельке - "Перевод займёт примерно 2 минуты" 
никогда такого не видел) обычно либо сразу включает на "Отключить", либо иногда ещё встречаю где надо обождать и вижу надпись "Перевод займёт около минуты", а вот про 2 минуты не видал)
ха)  оно ещё и само через время сменило надпись с "2 минуты" на "около минуты"
а это откуда вообще инфа про время ожидания?  сам яндекс рапортует?

**Reply by NullVerdict** [reply]
Да
